lib/rspec/expectations/handler.rb

```diff
@@ -69,7 +69,7 @@ class NegativeExpectationHandler
       def self.handle_matcher(actual, initial_matcher, message=nil, &block)
         ExpectationHelper.with_matcher(self, initial_matcher, message) do |matcher|
           return ::RSpec::Matchers::BuiltIn::NegativeOperatorMatcher.new(actual) unless initial_matcher
-          !(does_not_match?(matcher, actual, &block) || ExpectationHelper.handle_failure(matcher, message, :failure_message_when_negated))
+          does_not_match?(matcher, actual, &block) || ExpectationHelper.handle_failure(matcher, message, :failure_message_when_negated)
```

Comment:
@myronmarston @JonRowe After digging in to this further with a clear head, I see where my confusion lies. `#does_not_match?` here correctly returns `true` (it's true it does not match) and I don't really know why the logic is then being inverted.

Running off of fresh master, this behavior confuses me:

```plaintext
> expect("foo").to eq "foo"
=> true   # Good!

> expect("foo").to eq "bar"
RSpec::Expectations::ExpectationNotMetError:
expected: "bar"
     got: "foo"

(compared using ==)
# Good!

> expect("foo").not_to eq "foo"
RSpec::Expectations::ExpectationNotMetError:
expected: value != "foo"
     got: "foo"

(compared using ==)
# Good!

> expect("foo").not_to eq "bar"
=> false
# Whaaaat? This doesn't seem right.
```

Shouldn't the last one be `true`, because "foo" is not equal to "bar"? :confused: Seems to screw up what we're trying to accomplish in this PR (parity between `match` and `match_when_negated`)